astropy/modeling/tests/test_parameters.py

```diff
@@ -459,6 +459,17 @@ def test__set_unit(self):
         param._set_unit(u.m, True)
         assert param.unit == u.m
 
+        # Force magnitude unit (mag=False)
+        with pytest.raises(ValueError) as err:
+            param._set_unit(u.ABmag, True)
+        assert str(err.value) == (
+            "This parameter does not support the magnitude units such as mag(AB)"
+        )
```

Comment:
```suggestion
        with pytest.raises(ValueError, match="This parameter does not support the magnitude units such as mag(AB)"):
            param._set_unit(u.ABmag, True)
```

p.s. I would rather you not use `black` to get formatting like this:

```python
x = (
    some_expression
)
```

Reply:
Oh sorry, you need to do this:

```python
match=r"This parameter does not support the magnitude units such as mag\(AB\)")
```

Reply:
It does not like how the `mag(AB)` string is formatted. Replacing that with a wild card fixes it.